jade/wsgi/util.py

```diff
@@ -0,0 +1,45 @@
+from functools import wraps
+
+import flask
+
+from . import responses
+from .. import errors
+
+
+def authorized_user_action(config, trusted_clients, centralauth, rights_group):
+    """
+    Wrap a flask route and performs checks that a user is authorized and has
+    the rights to perform a specific action.  Authorization is confirmed via
+    OAuth or via a trusted MediaWiki installation and provided
+    a `global_user_id`.  If neither, return an authorization error.
+    """
+    def decorator(route):
+        @wraps(route)
+        def authorized_route(*args, **kwargs):
+            if 'authorization_key' in flask.request:
+                values = trusted_clients.authorize_and_decode(flask.request)
+                gu_id = values['gu_id']
+            elif 'mwoauth_access_token' in flask.session:
+                values = flask.request.values
+                gu_id = flask.session.get('mwoauth_identity')['id']
```

Comment:
I missed how we validated the access token here?

Reply:
We don't.  This will need to happen in connection with `mwoauth`.  Given that that is easy and we've done it multiple times, I've skipped spec'ing it out.

Reply:
Great.  Maybe just a comment here saying that it's not really authenticating, so we don't accidentally deploy like this?

Reply:
Ohh!  Even better,  I'll `raise NotImplementedError()`

Reply:
ty

Reply:
Just noting that this still isn't protected, should raise the NotImplementedError.